instrumentation/cats-effect-3.3.0/src/main/scala/cats/effect/IOFiber_Instrumentation.scala

```diff
@@ -0,0 +1,91 @@
+package cats.effect
+
+import cats.effect.internals.Utils._
+import cats.effect.tracing.RingBuffer
+import cats.effect.unsafe.IORuntime
+import com.newrelic.agent.bridge.AgentBridge
+import com.newrelic.api.agent.weaver.{NewField, Weave, Weaver}
+
+import java.util.concurrent.atomic.AtomicReference
+import java.util.logging.Level
+import scala.concurrent.ExecutionContext
+
+@Weave(originalName = "cats.effect.IOFiber")
+private final class IOFiber_Instrumentation[A](initLocalState: IOLocalState,
+                               cb: OutcomeIO[A] => Unit,
+                               startIO: IO[A],
+                               startEC: ExecutionContext,
+                               private[this] val runtime: IORuntime
+                              ) {
+
+  //possibly wrap startEc
+
+  @NewField
+  private val tokenAndRefCount: AtomicReference[AgentBridge.TokenAndRefCount] =
+    new AtomicReference(getThreadTokenAndRefCount(fiberRef))
+
+  incrementTokenRefCount(this.tokenAndRefCount.get())
+  logTokenInfo(tokenAndRefCount.get(), fiberRef, "IOFiber token info set")
+  private val IOEndFiber: IO.EndFiber.type = Weaver.callOriginal();
+
+  private def fiberRef: IOFiber[A] = this.asInstanceOf[IOFiber[A]]
+  def run(): Unit = {
```

Comment:
can you add a line between these functions?
reading thru this and getting confused by which is the function for this body.